File: code_contest.cpp
#include <algorithm>
#include <chrono>
#include <deque>
#include <fstream>
#include <iomanip>
#include <iostream>
#include <limits>
#include <list>
#include <map>
#include <queue>
#include <random>
#include <set>
#include <sstream>
#include <stack>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <vector>

using namespace std;

#define rep(i, a, b) for (int i = int(a); i < int(b); ++i)
#define rrep(i, a, b) for (int i = int(a); i >= int(b); --i)
#define all(x) begin(x), end(x)
#define sz(x) (int)(x).size()
#define mp make_pair
//#define f first
//#define s second
#define getunique(v)                                  \
    {                                                 \
        sort(v.begin(), v.end());                     \
        v.erase(unique(v.begin(), v.end()), v.end()); \
    }

#define caf const auto&

using lld = long double;
using ull = uint64_t;
using ll = int64_t;

using Index = pair<int, int>;

using pii = pair<int, int>;
using pll = pair<ull, ull>;
using pdd = pair<double, double>;
using vi = vector<int>;
using vvi = vector<vector<int> >;
using vvvi = vector<vector<vector<int> > >;

using vb = vector<bool>;
using vvb = vector<vector<bool> >;
using vvvb = vector<vector<vector<bool> > >;

double dmax = numeric_limits<double>::max();
int imax = numeric_limits<int>::max();

// template <typename num_t>
// using ordered_set = tree<num_t, null_type, less<num_t>, rb_tree_tag,
//                          tree_order_statistics_node_update>;

// benq - print any container + char_pair
template<typename T, typename = void>
struct is_iterable : false_type {
};

template<typename T>
struct is_iterable<T, void_t<decltype(begin(declval<T>())), decltype(end(declval<T>()))> > : true_type {
};

template<typename T>
typename enable_if<is_iterable<T>::value && !is_same<T, string>::value, ostream &>::type
operator<<(ostream &cout, T const &v);

template<typename A, typename B>
ostream &operator<<(ostream &cout, pair<A, B> const &p) {
    return cout << "(" << p.f << ", " << p.s << ")";
}

template<typename T>
typename enable_if<is_iterable<T>::value && !is_same<T, string>::value, ostream &>::type
operator<<(ostream &cout, T const &v) {
    cout << "[";
    for (auto it = v.begin(); it != v.end();) {
        cout << *it;
        if (++it != v.end()) cout << ", ";
    }
    return cout << "]";
}

template<typename A, typename B>
istream &operator>>(istream &cin, pair<A, B> &p) {
    cin >> p.first;
    return cin >> p.second;
}

mt19937_64 rng(std::chrono::steady_clock::now().time_since_epoch().count());
// mt19937_64 rng(61378913);
/* usage - just do rng() */

// use for atcoder
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wunused-result"

void usaco(string filename) {
    freopen((filename + ".in").c_str(), "r", stdin);
    freopen((filename + ".out").c_str(), "w", stdout);
}

#pragma GCC diagnostic pop

vector<string> ReadInput() {
    vector<string> rst;

    std::string line;
    while (std::getline(std::cin, line)) {
        if (line.empty()) continue;
        rst.push_back(line);
    }

    return rst;
}

vector<string> ReadAllFileLines(const string &filename) {
    auto rst = vector<string>{};

    std::ifstream infile(filename);
    std::string line;
    while (std::getline(infile, line)) {
        rst.push_back(line);
    }
    return rst;
}

template<typename T>
string Join(const vector<T> &input, const string &split) {
    stringstream ss;
    for (size_t i = 0; i < input.size(); ++i) {
        if (i != 0) ss << split;
        ss << input[i];
    }

    return ss.str();
}

template<typename T>
vector<T> SplitLine(const string &s) {
    vector<T> rst;

    T t;
    stringstream ss(s);
    while (ss >> t) {
        rst.push_back(t);
    }

    return rst;
}

template<typename T>
vector<T> SplitLine(int n, const string &s) {
    vector<T> rst(n);

    stringstream ss(s);
    for (int i = 0; i < n; i++) {
        ss >> rst[i];
    }

    return rst;
}

// #include <atcoder/all>
// using namespace atcoder;

const lld pi = 3.14159265358979323846;
// const ll mod = 1000000007;
// const ll mod = 998244353;
// ll mod;

// ll n, m, q, k, l, r, x, y, z;
// const ll template_array_size = 1e6 + 17646;
// string s, t;


struct Solve {
    auto Solution(const vector<string> &lines) {
        cout << "cpp" << endl;
        return lines;
    }
};

/*
ID: racer
TASK: sprime
LANG: C++
*/

int main() {
#ifdef LOCAL
    usaco("test");
    auto begin = std::chrono::high_resolution_clock::now();
#else
    usaco("sprime");
#endif

    caf lines = ReadInput();
    caf rst = Solve().Solution(lines);
    for (caf v: rst) {
        cout << v << "\n";
    }

#ifdef LOCAL
    auto end = std::chrono::high_resolution_clock::now();
    cerr << setprecision(4) << fixed;
    cerr << "Execution time: " << std::chrono::duration_cast<std::chrono::duration<double> >(end - begin).count()
            << " seconds" << endl;
#endif
}

File: code_contest.fs
open System
open System.Collections.Generic
open System.IO
open Microsoft.FSharp.Collections

type Index = int * int
type ll = int64
type ull = uint64
type pii = int * int
type pll = int64 * int64
type vi = int[]
type vvi = int[][]
type vvvi = int[][][]

type vb = bool[]
type vvb = bool[][]
type vvvb = bool[][][]

let IMIN = Int32.MinValue
let IMAX = Int32.MaxValue
let DMAX = Double.MaxValue
let DMIN = Double.MinValue
let DNAN = Double.NaN
let pi: double = 3.14159265358979323846

let tmp_dict = Dictionary<int, int>()
let tmp_set = HashSet<int>()

#if DEBUG
let RedirectConsole () =
    let if_stream = new FileStream("test.in", FileMode.OpenOrCreate, FileAccess.Read)
    let of_stream = new FileStream("test.out", FileMode.Create, FileAccess.Write)
    let sin = new StreamReader(if_stream)
    let sout = new StreamWriter(of_stream)
    sout.AutoFlush <- true

    Console.SetIn(sin)
    Console.SetOut(sout)
#endif

let ReadInputV2 () : string[] =
    let rec loop acc =
        let line = Console.ReadLine()
        if line <> null then loop (line :: acc) else acc

    loop [] |> List.rev |> List.toArray

let ReadInput () : string[] =
    let input = ResizeArray<string>()
    let mutable line = Console.ReadLine()

    while line <> null do
        input.Add(line)
        line <- Console.ReadLine()

    input |> Seq.toArray


// let Join<'t> (src: 't []) : string =
//     src |> Array.map string |> String.concat " "

let Split2Str (src: string) : string[] = src.Split(" ")
let Split2Int (src: string) : int[] = src |> Split2Str |> Array.map int

module Solve =
    let Solution (lines: string[]) =
        let x = ref 123
        let y = ref "234"
        y.Value <- "345"
        let z: bool[] = Array.empty
        printfn ($"z={z |> Array.forall id}")

        let test_arr = ResizeArray<int>()
        test_arr.Add(123)

        let test_arr2 = test_arr
        test_arr2.Add(345)

        test_arr |> Seq.iter (printfn "%d")

        Array.append [| "from F#" |] lines


[<EntryPoint>]
let main _ =
#if DEBUG
    RedirectConsole()
    let start = DateTime.Now
#endif
    ReadInput() |> Solve.Solution |> Seq.iter (printfn "%s")

#if DEBUG
    printfn $"Execution time: %A{(DateTime.Now - start).TotalSeconds} seconds"
#endif

    0
